File: easy/1800_maximum_ascending_subarray_sum/solution.py
from typing import List


class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        curr_ascending_sum = max_ascending_sum = 0

        prev = 0

        for num in nums:
            if prev < num:
                curr_ascending_sum += num
            else:
                max_ascending_sum = max(curr_ascending_sum, max_ascending_sum)
                curr_ascending_sum = num

            prev = num

        return max(curr_ascending_sum, max_ascending_sum)

File: easy/1800_maximum_ascending_subarray_sum/solution.cpp
using namespace std;

#include <vector>

class Solution {
public:
  int maxAscendingSum(vector<int> &nums) {
    const size_t numsLen = nums.size();

    int currSum = nums[0];
    int maxSum = 0;

    for (size_t i = 1; i < numsLen; ++i) {
      if (nums[i - 1] < nums[i]) {
        currSum += nums[i];
      } else {
        maxSum = max(currSum, maxSum);
        currSum = nums[i];
      }
    }

    return max(currSum, maxSum);
  }
};
